### PR TITLE
bpo-37207: Add _PyArg_NoKwnames helper function.

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue37207

### CODE REVIEW

File: Python/getargs.c [hunk 1]
```diff
@@ -2831,6 +2831,24 @@ _PyArg_NoPositional(const char *funcname, PyObject *args)
     return 0;
 }
 
+int
+_PyArg_NoTupleKeywords(const char *funcname, PyObject *kwnames)
```
Comment (corona10) [inline Python/getargs.c]: @vstinner 
But this is essentially same implemenation of `_PyArg_NoStackKeywords`which is removed.

File: Python/getargs.c [hunk 2]
```diff
@@ -2831,6 +2831,24 @@ _PyArg_NoPositional(const char *funcname, PyObject *args)
...
+        return 1;
+    }
+
+    assert(PyTuple_CheckExact(kwnames));
+
+    if (PyTuple_GET_SIZE(kwnames) == 0) {
+        return 1;
+    }
+
+    PyErr_Format(PyExc_TypeError, "%.200s() takes no keyword arguments",
```
Comment (vstinner) [inline Python/getargs.c]: The %.200s is an old habbit from the 90s which is no longer needed. PyErr_Format() is able to allocate enough memory for longer function name.

```suggestion
    PyErr_Format(PyExc_TypeError, "%s() takes no keyword arguments",
```

File: Include/modsupport.h [hunk 3]
```diff
@@ -61,10 +61,13 @@ PyAPI_FUNC(int) _PyArg_UnpackStack(
 
 PyAPI_FUNC(int) _PyArg_NoKeywords(const char *funcname, PyObject *kwargs);
 PyAPI_FUNC(int) _PyArg_NoPositional(const char *funcname, PyObject *args);
+PyAPI_FUNC(int) _PyArg_NoTupleKeywords(const char *funcname, PyObject *kwnames);
```
Comment (vstinner) [inline Include/modsupport.h]: Bikeshedding!

```suggestion
PyAPI_FUNC(int) _PyArg_NoKwnames(const char *funcname, PyObject *kwnames);
```

**Comment by corona10** [comment]
@vstinner Thanks Victor Updated!

**Comment by vstinner** [comment]
Perfect. I have nothing to add.

**Pull request merged**